Add support for iceberg set properties

pratyakshsharma opened the pull request:
## Description
Add support for Iceberg ALTER TABLE SET PROPERTIES
Needs to merge after https://github.com/prestodb/presto/pull/21495 and https://github.com/prestodb/presto/pull/21250.

Docs and testing is pending.

## Motivation and Context
https://github.com/prestodb/presto/issues/21251

## Impact
```ALTER TABLE foo SET PROPERTIES (commit_retries = 5);```

## Test Plan
Unit testing (pending) and manual testing (done)

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Iceberg Changes
* Add implementation for ALTER TABLE SET PROPERTIES command
```

pratyakshsharma commented:
Please do not review it for now, it needs to merge after https://github.com/prestodb/presto/pull/21495 and https://github.com/prestodb/presto/pull/21250.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -644,6 +649,11 @@ The table is partitioned by the transformed value of the column::
 
      ALTER TABLE iceberg.web.page_views ADD COLUMN ts timestamp WITH (partitioning = 'hour');
 
+Table properties can be modified for an iceberg table using ALTER TABLE SET PROPERTIES statement. Only `commit_retries` can be modified at present.
```
```suggestion
Table properties can be modified for an Iceberg table using an ALTER TABLE SET PROPERTIES statement. Only `commit_retries` can be modified at present.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -644,6 +649,11 @@ The table is partitioned by the transformed value of the column::
      ALTER TABLE iceberg.web.page_views ADD COLUMN ts timestamp WITH (partitioning = 'hour');
 
+Table properties can be modified for an iceberg table using ALTER TABLE SET PROPERTIES statement. Only `commit_retries` can be modified at present.
+For example, to set commit_retries to 6 for table `iceberg.web.page_views_v2`, use the below statement::
```
```suggestion
For example, to set `commit_retries` to 6 for the table `iceberg.web.page_views_v2`, use::
```

steveburnett reviewed (changes requested):
Thanks for the doc! A couple of small suggestions - if they don't look good to you, let me know and we'll figure it out.

pratyakshsharma commented:
Thank you for your review @steveburnett . I am closing the PR and these changes are now added to https://github.com/prestodb/presto/pull/21495 as a separate commit. I have addressed your review comments there itself. 
Please have a look and let us continue the discussion there. 
cc @yingsu00
